src/engraving/dom/spanner.cpp

```diff
@@ -992,7 +992,8 @@ Chord* Spanner::endChord()
 ChordRest* Spanner::startCR()
 {
     assert(m_anchor == Anchor::SEGMENT || m_anchor == Anchor::CHORD);
-    if (!m_startElement || m_startElement->score() != score()) {
+    if (!m_startElement || m_startElement->score() != score()
+        || (type() == ElementType::LYRICSLINE && m_startElement->track() != track())) {
```

Comment:
This feels a bit hacky. Ideally, startCR() should be const and only return the start element, never compute it. The correct start element should be computed and set earlier, if possible, as soon as the spanner is created. Maybe we can find the function that creates the lyrics line and set the correct start chord from there? Otherwise, Spanner::computeStartElement could also be a good place to put this.

Reply:
Agreed.  `Spanner::computeStartElement` is called a few times during layout, so it should probably go there - any startElement we set will get overwritten by this.